.github/workflows/precommit.yaml

```diff
@@ -16,13 +18,19 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     timeout-minutes: 10
+    strategy:
+      matrix:
+        python_version:
+          - "3.9"
```

Comment:
It looks like the `install` workflow runs on Python 3.7, 3.8 and 3.9.  Why does this only run on 3.9?

Reply:
We want to make sure that the streaming installation works as expected on 3.7, 3.8 and 3.9. However, we are running the pre-commit workflow on the latest python to make sure it passes since it is just used for linting. Making it to run on a multiple python version is just adding more redundancy without having much benefit.

Reply:
I think pyright has different linting rules per Python version so technically it isn't redundant.  It might be overkill though so let's keep it minimal for now and we can add additional versions if necessary.